### PR TITLE
Add RouteRootGuard, prevent user from going to empty routes

### PR DESCRIPTION
**Pull request by Filip-A25**

### CODE REVIEW

File: src/modules/global/components/RouteRootGuard.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+interface Props extends PropsWithChildren {
+  isErrorPage?: boolean;
+}
+
+export const RouteRootGuard = ({ children, isErrorPage }: Props) => {
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loggedState);
+  const setUserData = useSetRecoilState(userState);
+  const [isPageLoading, setIsPageLoading] = useState(true);
+
+  useEffect(() => {
```
Comment (StjepanDzojic) [inline src/modules/global/components/RouteRootGuard.tsx]: Remove this because you have the provider now.

File: src/App.tsx [hunk 2]
```diff
@@ -9,30 +9,36 @@ import {
...
-          path="/home/*"
-          element={
-            <RoutePrivateGuard>
-              <Configurator />
-            </RoutePrivateGuard>
-          }
-        />
-      </Routes>
-    </div>
+    <FirebaseAuthProvider>
```
Comment (StjepanDzojic) [inline src/App.tsx]: Use the provider inside index.tsx

File: src/modules/global/components/RouteRootGuard.tsx (hunk 1 above)
Comment (Filip-A25) [reply src/modules/global/components/RouteRootGuard.tsx]: Done

File: src/App.tsx (hunk 2 above)
Comment (Filip-A25) [reply src/App.tsx]: Done

File: src/modules/global/components/RouteRootGuard.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export const RouteRootGuard = ({ children, isErrorPage }: Props) => {
+  const isLoggedIn = useRecoilValue(loggedState);
+  const isPageLoading = useRecoilValue(pageLoadingState);
+
+  if (isPageLoading) return <PageLoading />;
+
+  if (!isLoggedIn && !isErrorPage) return <Navigate to="/auth/log-in" />;
+  if (!isLoggedIn && isErrorPage)
+    return (
+      <NotFoundPage returnPageTitle="sign in" returnPagePath="/auth/log-in" />
```
Comment (StjepanDzojic) [inline src/modules/global/components/RouteRootGuard.tsx]: You could have used the routes object.

Comment (Filip-A25) [reply src/modules/global/components/RouteRootGuard.tsx]: Done

**Review by StjepanDzojic — APPROVED** [review]

**Pull request merged**